File: src/main/java/com/github/frcsty/frozenactions/actions/broadcast/SoundBroadcastAction.java
package com.github.frcsty.frozenactions.actions.broadcast;

import com.github.frcsty.frozenactions.actions.Action;
import com.github.frcsty.frozenactions.actions.ActionContext;
import org.bukkit.Bukkit;
import org.bukkit.Sound;
import org.jetbrains.annotations.NotNull;

public final class SoundBroadcastAction implements Action {

    @Override
    public @NotNull String getId() {
        return "SOUNDBROADCAST";
    }

    @Override
    public void run(final @NotNull ActionContext context) {
        final String[] args = context.getDataAsStringArray(";");

        final Sound sound = Sound.valueOf(args[0]);
        final float volume = Float.parseFloat(args[1]);
        final float pitch = Float.parseFloat(args[2]);

        Bukkit.getOnlinePlayers().forEach(player -> player.playSound(player.getLocation(), sound, volume, pitch));
    }
}

File: src/main/kotlin/com/github/frcsty/frozenactions/actions/broadcast/SoundBroadcastAction.kt
package com.github.frcsty.frozenactions.actions.broadcast

import com.github.frcsty.frozenactions.actions.Action
import org.bukkit.Bukkit
import org.bukkit.Sound
import org.bukkit.entity.Player

object SoundBroadcastAction : Action {

    override val id = "SOUNDBROADCAST"

    override fun run(player: Player, data: String) {
        val args = data.split(";")
        val sound = Sound.valueOf(args[0])
        val volume: Float = Float.fromBits(args[1].toInt())
        val pitch: Float = Float.fromBits(args[2].toInt())

        Bukkit.getServer().onlinePlayers.forEach { it.playSound(it.location, sound, volume, pitch) }
    }
}